core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -181,7 +184,8 @@ private void writeRaw(byte[] b, int off, int len) {
         commitToSink(false);
       }
       if (buffer == null) {
-        buffer = bufferAllocator.allocate(maxFrameSize);
+        // Allocate as big a buffer as the allocator will allow
```

Comment:
Not exactly no, see discussion below. Renamed param in interface to 'capacityHint'